test/FunctionalTests/Microsoft.OData.Client.Tests/ClientEdmModelTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ClientEdmModelTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using Microsoft.OData.Edm;
+using Xunit;
+
+namespace Microsoft.OData.Client.Tests
+{
+    public class ClientEdmModelTests
+    {
+        /// <summary>
+        /// A test to show that an abstract complex type with no properties defined is 
+        /// assignable from a concrete complex type that extends it.
+        /// Product is the abstract complex type with no properties defined and
+        /// ProductA is the concrete complex type that extends the abstract 
+        /// complex type(Product) with no properties defined. 
+        /// </summary>
+        [Fact]
+        public void BaseTypeOfProductAIsProduct()

```

Comment:
I'd suggest 2 tests to show that the empty abstract works for ComplextTypes but fails for EntityTypes

Reply:
it should not fail for entity types. You can have an empty abstract type for entity types.

Reply:
Oh right. My mistake